python/lsst/sims/alertsim/avro_utils.py

```diff
@@ -38,28 +39,54 @@ def catsim_to_avro(list_of_query_dicts, schemaURI='avsc/diasource.avsc'):
     writing_time = timer()
     writer = DataFileWriter(open("avsc/alert.avro", "wb"), DatumWriter(), alert_schema)
 
-    for qd in list_of_query_dicts:
+    print("number of events %d" % (len(list_of_alert_dicts)))
 
-        alert_dict = {'alertID':45135, 'l1dbID':12545, 'diaSource':qd, 'prevDiaSources':[qd]*30}
-        #alert_dict = {'alertID':45135, 'l1dbID':12545, 'diaSource':qd}
-        json_qd = json.loads(json.dumps(alert_dict))
-        print(json_qd)
-        writer.append(json_qd)
+    alert_dicts_divided = defaultdict(list)
+    for d in list_of_alert_dicts:
+        alert_dicts_divided[(d['diaSource']['ccdVisitId'])].append(d)
+    
+    print("##### starting serialization")
+
+    for ix, list_per_chip in enumerate(alert_dicts_divided.values()):
+        print("Ccd and visit id %d" % (list_per_chip[0]['diaSource']['ccdVisitId']))
+        #last 4 digits of ccdVisitId are chip number
+        chipNum = str(list_per_chip[0]['diaSource']['ccdVisitId'])[-4:]
+
+        serialization_timer = timer()
+
+        #open file with a name of a chipNum in append mode
+        file_obj = open("json_output/"+session_dir+"/"+chipNum, 'a')
+        
+        avro_validated = False
+
+        for alert_dict in list_per_chip:
+            json_qd = json.loads(json.dumps(alert_dict))
+        
+            json.dump(json_qd, file_obj)
```

Comment:
Can you not just call `json.dump(alert_dict, file_obj)`?

Reply:
That wouldn't work earlier. Now looks fine but I ran into other issues - incompatibility of avro with py3 and json not willing to serialize numpy types. I think I managed to solve these, please look into it when I make a new pull req.